### PR TITLE
Display note history as a sidebar treeview

### PR DESCRIPTION
**Pull request by JamesCamel**
* Add user login mechanism.
* Open textdoucment as diffeternt browsing mode when clicking buttons on sidebar treeview.
* Create new note from active editor by vscode command. (Copy the note URL to clipboard and open it on browser)
* Add enterprise and serverurl configuration setting.

### CODE REVIEW

File: src/api.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import * as apiClient from '@hackmd/api';
+export const apiExportType = apiClient.ExportType;
+export const API = new apiClient.default();
```
Comment (Yukaii) [inline src/api.ts:3]:
```javascript
import APIClient from '@hackmd/api';

// in other files want to use ExportType
import { ExportType } from '@hackmd/api'
```

**Comment by Yukaii** [comment]
`._DS_Store` is not added to .gigitnore and accidentally committed.

File: src/api.ts (hunk 1 above)
Comment (JamesCamel) [reply src/api.ts]: Fixed.

File: package.json [hunk 2]
```diff
@@ -36,12 +41,109 @@
 	],
 	"main": "./dist/extension.js",
 	"contributes": {
+		"configuration": {
+			"title": "HackMD",
+			"properties": {
+				"Hackmd.serverURL": {
+					"type": "string",
+					"default": "https://hackmd.io/",
```
Comment (Yukaii) [inline package.json:49]: Default should be `https://hackmd.io` now. (wihout trailing slash) 

upstream: https://github.com/hackmdio/api-client/issues/5

**Comment by Yukaii** [comment]
Big thanks to @JamesCamel

**Pull request merged**